dhall/src/Dhall.hs

```diff
@@ -94,14 +94,14 @@ module Dhall
     , Inject
     , inject
     , genericToDhall
-    , RecordInputType(..)
-    , inputFieldWith
-    , inputField
-    , inputRecord
-    , UnionInputType(..)
-    , inputConstructorWith
-    , inputConstructor
-    , inputUnion
+    , RecordEncoder(..)
+    , encodeFieldWith
+    , encodeField
+    , encodeRecord
+    , UnionEncoder(..)
+    , encodeConstructorWith
+    , encodeConstructor
+    , encodeUnion
```

Comment:
I think it was good to get rid of the "input*" names here, but I'm not convinced about these "encode*" names.